scandium-core/src/main/java/org/eclipse/californium/scandium/dtls/ServerHandshaker.java

```diff
@@ -336,7 +336,13 @@ private void receivedClientCertificate(final CertificateMessage message) throws
 		clientCertificate = message;
 		verifyCertificate(clientCertificate);
 		clientPublicKey = clientCertificate.getPublicKey();
-		peerCertPath = message.getCertificateChain();
+		peerCertPath = clientCertificate.getCertificateChain();
+		if (clientAuthenticationRequired && clientPublicKey == null) {
+			LOGGER.debug("Certificate validation failed: x509 could not be trusted!");
+			AlertMessage alert = new AlertMessage(AlertLevel.FATAL, AlertDescription.HANDSHAKE_FAILURE,
+					session.getPeer());
+			throw new HandshakeException("Client Certificate required!", alert);
+		}
```

Comment:
Should we do `verifyCertificate(clientCertificate);` after this new check ?

Reply:
Done-